Topics/Development_Process/Build_Requirements/Introduction_to_make_and_Makefiles.md

```diff
@@ -0,0 +1,228 @@
+# Introduction to ```make``` and Makefiles
+
+## Table of contents
+
+### [Introduction](#introduction-1)
+
+### [Basic Makefile Characteristics](#basic-makefile-characteristics-1)
+
+### [Making a Makefile](#making-a-makefile-1)
+
+### [Advantages and Disadvantages to ```make``` and Makefiles](#advantages-and-disadvantages-to-make-and-makefiles-1)
+
+### [Additional Resources](#additional-resources-1)
+
+## Introduction
+
+#### What are ```make``` and Makefiles
+```make``` is a software build automation tool that builds programs and libraries, and determines what needs to be recompiled. It accomplishes this by using Makefiles that specify compilation targets and link their dependencies, and checking their last modified dates during compilation.
+
+## Basic Makefile Characteristics
+
+#### Target
+
+A target is the name of a task to be executed. It is typically the name of some executable or object files to be generated. A phony target is a target not named after a file.
+
+#### Prerequisite
+
+Prerequisites are files or tasks that the target depends on as inputs to create the target. If prerequisites have been changed since the last build, the target will be considered outdated and will need to be rebuilt.
+
+#### Recipe
+
+A recipe is an action that ```make``` will execute for the target it is under. It can be one or more commands, but each single recipe line must be preceded by a tab character[^1]
+
+#### Rule
+
+Rules put the above together and link together the prerequisites to the target and define the recipes ```make``` will carry out for the designated target. By default, the first target of the first rule in the Makefile will be run by ```make```, other rules would have to be explicitly referenced by ```make target-name```.
+
+#### Variables
+
+Variables store values to be used throughout the file that will be substituted in whenever the variable is referenced.
+
+
+#### Functions
+
+Functions can be called to process some argument(s) and substitute some text similar to a variable. These functions are predefined by ```make```.
+
+## Making a Makefile
+
+#### Running a Simple Makefile
+
+The following will be written in a file called "Makefile", because make will automatically look for a file of that name in the current directory[^2].  
```

Comment:
I notice that you formatted “make” to be in code blocks, but I think you missed it here.

Reply:
Oh I didn't catch that, thanks!